libs/market-list/src/lib/use-markets.ts

```diff
@@ -0,0 +1,110 @@
+import { useYesterday } from '@vegaprotocol/react-helpers';
+import * as Schema from '@vegaprotocol/types';
+import isEqual from 'lodash/isEqual';
+import { useEffect, useMemo } from 'react';
+import type {
+  MarketHookDataFragment,
+  MarketHookQuery,
+  MarketsDataHookSubscription,
+  MarketsDataHookSubscriptionVariables,
+} from './__generated__/MarketsHooks';
+import {
+  MarketsDataHookDocument,
+  useMarketsHookQuery,
+} from './__generated__/MarketsHooks';
+
+export const useMarkets = () => {
+  const yesterday = useYesterday();
+  const { data, loading, error, subscribeToMore } = useMarketsHookQuery({
+    variables: {
+      since: new Date(yesterday).toISOString(),
+      interval: Schema.Interval.INTERVAL_I1H,
+    },
+  });
+
+  const ids = useMemo(() => {
+    if (!data?.marketsConnection?.edges.length) return [];
+    return data.marketsConnection.edges.map((e) => e.node.id);
+  }, [data]);
+
+  useEffect(() => {
+    const unsub = subscribeToMore<
+      MarketsDataHookSubscription,
+      MarketsDataHookSubscriptionVariables
+    >({
+      document: MarketsDataHookDocument,
+      variables: { marketIds: ids },
+      updateQuery: (prev, { subscriptionData }) => {
```

Comment:
maybe we will need some throttling here? Update can happen many times per second